challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map {|i| i.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select {|i| i == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject {|i| i == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep(/\Ac|\Ad/)
```

Comment:
nice case for a regex.  Maybe `/\A[cd]/`?

Check out the ruby string methods, though, and see if you can find one that is fine-tuned for this scenario?